Add mixer knob reading/writing support to snd

gerow opened the pull request:
This defines the interface between snd and the audio driver and begins defining some of the pieces for the ioctl calls between userspace and snd. The userspace interface isn't fully realized yet, but this patch series includes three debug shell functions to play around with changing the volume using only the interface to the sound card driver:

snd_off: turns volume all the way down
snd_half: turns volume to half
snd_full: turns volume to full

klange commented on Makefile:
```diff
@@ -188,7 +188,7 @@ kernel/sys/version.o: kernel/*/*.c kernel/*.c
 hdd/mod/%.ko: modules/%.c ${HEADERS}
 	@${BEG} "CC" "$< [module]"
-	@${CC} -T modules/link.ld -I./kernel/include -nostdlib ${CFLAGS} -c -o $@ $< ${ERRORS}
+	@${CC} -T modules/link.ld -I./kernel/include -I./userspace -nostdlib ${CFLAGS} -c -o $@ $< ${ERRORS}
```
Plop your system headers in `userspace/lib` and then symlink them in `kernel/include/mod`, rather than having the modules pull in the include path. Keeps things nice and separate.

gerow commented:
When I rebased and removed that commit your comment disappeared. That was certainly unexpected. Anyway, I've gone ahead and moved things around as you suggested: move userspace snd.h to sound.h and just symlink sound.h into the module directory.

I need to figure out a better way to change code in a patch series without github getting angry at me...

klange commented:
Rebasing is fine, github actually manages it reasonably well if you look above.

It looks like you're intending to implement knob querying through ioctls; we should get in that in before merging.

gerow commented:
sounds good to me. I'll probably just add that to this branch and re-ping the pull request.

gerow commented:
Added the ioctl interface at /dev/mixer and a command line utility for interacting with it. Also rebased from latest commit in strawberry-dev. Please take another look.
